src/test/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctionsTest.scala

```diff
@@ -185,4 +191,199 @@ class PipelineFunctionsTest extends AnyFunSpec with DataFrameComparer with Spark
     }
   }
 
-}
\ No newline at end of file
+  /**
+   * Below tests are added as a part of OV-43
+   */
+
+  describe("Tests for parseEHConnectionString") {
+    it("should throw an exception - empty string") {
+      Given("an event hub connection string")
+      val ehConnString = ""
+
+      When("the connection string is not in correct format")
+
+      Then("the function throws an exception")
+      assertThrows[BadConfigException] (PipelineFunctions.parseEHConnectionString(ehConnString))
+    }
+    it("should throw an exception - incorrect format") {
```

Comment:
sometimes the SharedAccessKey has a + and sometimes it doesn't. We've had cases where the + broke our parser, can we add a test case for that to ensure we don't hit that again? Additionally, the sharedAccessKey format is published, perhaps we can test the edge cases of that format to help guard against future breaks. This is a very nasty and public facing break if it happens.